common/djangoapps/track/views/segmentio.py

```diff
@@ -123,16 +123,16 @@ def track_segmentio_event(request):  # pylint: disable=too-many-statements
     full_segment_event = request.json
 
     # We mostly care about the properties
-    segment_properties = full_segment_event.get('properties', {})
+    segment_properties = _get_dict_value_with_default(full_segment_event, 'properties', {})
 
     # Start with the context provided by Segment in the "client" field if it exists
     # We should tightly control which fields actually get included in the event emitted.
-    segment_context = full_segment_event.get('context', {})
+    segment_context = _get_dict_value_with_default(full_segment_event, 'context', {})
 
     # Build up the event context by parsing fields out of the event received from Segment
     context = {}
 
-    library_name = segment_context.get('library', {}).get('name')
+    library_name = _get_dict_value_with_default(segment_context, 'library', {}).get('name')
```

Comment:
From the DESUPPORT ticket:

> You probably want to replace .get('name') with  .get('name', '') (or using the new helper method) as well, or it will continue to fail.

Is that still true?

Reply:
Thanks for following up on that, but no, that is not true.  The library name is not a dict but just a value, and it is compared against an expected value (or values).   So whether it is NULL or a zero-length string does not matter.

Reply:
Agreed that this is fine.  I added this note for a different reason that I was wrong about. :)  I didn't realize that get on a dict doesn't raise an exception if the key is missing.
```
>>> x = {}
>>> print({}.get('name'))
None
```